### PR TITLE
Specify the PATH_CHALLENGE timer period

### LINKED ISSUE #2910: PATH_CHALLENGE timer period
Section 9.4 of draft-ietf-quic-transport:
   An endpoint might set a
   separate timer when a PATH_CHALLENGE is sent, which is cancelled when
   the corresponding PATH_RESPONSE is received.  If the timer fires
   before the PATH_RESPONSE is received, the endpoint might send a new
   PATH_CHALLENGE, and restart the timer for a longer period of time.

GF: Possibly so, but what value has this timeout? 
- To avoid potential congestion collapse, I think this needs to be conservative enough to deal with paths with significant RTT variance, or where there is a change of base RTT. 1 sec, then 3 seconds, and then exponentially backed-off would be expected for TCP SYNs, and I suggest could be appropriate.

### PR DESCRIPTION
**Pull request by ianswett**
Fixes #2910

### CODE REVIEW

File: draft-ietf-quic-transport.md [hunk 1]
```diff
@@ -2176,9 +2176,10 @@ path is no longer needed (such as the case in {{off-path-forward}}).
 A sender can make exceptions for probe packets so that their loss detection is
 independent and does not unduly cause the congestion controller to reduce its
 sending rate.  An endpoint might set a separate timer when a PATH_CHALLENGE is
-sent, which is cancelled when the corresponding PATH_RESPONSE is received.  If
-the timer fires before the PATH_RESPONSE is received, the endpoint might send a
-new PATH_CHALLENGE, and restart the timer for a longer period of time.
+sent, which is cancelled if the corresponding PATH_RESPONSE is received or a
+new PATH_CHALLENGE is sent if it expires before receiving a PATH_RESPONSE.
+This timer SHOULD be set as described in section 5.3 of {{QUIC-RECOVERY}} and
```
Comment (martinthomson) [inline draft-ietf-quic-transport.md]:
```suggestion
This timer SHOULD be set as described in Section 5.3 of {{QUIC-RECOVERY}} and
```

File: draft-ietf-quic-transport.md [hunk 2]
```diff
@@ -2176,9 +2176,10 @@ path is no longer needed (such as the case in {{off-path-forward}}).
 A sender can make exceptions for probe packets so that their loss detection is
 independent and does not unduly cause the congestion controller to reduce its
 sending rate.  An endpoint might set a separate timer when a PATH_CHALLENGE is
-sent, which is cancelled when the corresponding PATH_RESPONSE is received.  If
-the timer fires before the PATH_RESPONSE is received, the endpoint might send a
-new PATH_CHALLENGE, and restart the timer for a longer period of time.
+sent, which is cancelled if the corresponding PATH_RESPONSE is received or a
```
Comment (janaiyengar) [inline draft-ietf-quic-transport.md]:
```suggestion
sent, which is cancelled if the corresponding PATH_RESPONSE is received. If
```

File: draft-ietf-quic-transport.md [hunk 3]
```diff
@@ -2176,9 +2176,10 @@ path is no longer needed (such as the case in {{off-path-forward}}).
...
+new PATH_CHALLENGE is sent if it expires before receiving a PATH_RESPONSE.
```
Comment (janaiyengar) [inline draft-ietf-quic-transport.md]:
```suggestion
the timer fires before the PATH_RESPONSE is received, the endpoint might send a
new PATH_CHALLENGE, and restart the timer for a longer period of time. 
```

File: draft-ietf-quic-transport.md (hunk 2 above)
Comment (janaiyengar) [inline draft-ietf-quic-transport.md]: It's not clear what the "or" applies to.  You can undo the changes and simply add your new sentence at the end; see my suggestions below.

Comment (ianswett) [reply draft-ietf-quic-transport.md]: Agreed, that was awkward, thanks for the suggestions.  Accepted.
